04.ls/wc.rb

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  argv = define_option_boolean[1]
+  if argv.empty?
+    inputs = $stdin.readlines
+    formatted_inputs = inputs.join
+    count = { line: 0, word: 0, byte: 0 }
+    count[:line] = count_lines(formatted_inputs)
+    count[:word] = count_words(formatted_inputs)
+    count[:byte] = formatted_inputs.size
+    formatted_lwc = format_lwc(**count)
+    output_standard(formatted_lwc)
+  else
+    output_file(make_lwc)
+  end
+end
+
+def define_option_boolean
+  opt_params = { l: false, w: false, c: false }
+  opt = OptionParser.new
+
+  opt.on('-l') { |v| opt_params[:l] = v }
+  opt.on('-w') { |v| opt_params[:w] = v }
+  opt.on('-c') { |v| opt_params[:c] = v }
+  argv = opt.parse(ARGV)
+  [opt_params, argv]
+end
+
+def make_lwc
+  argv = define_option_boolean[1]
+  lwc_data = []
+  argv.each do |file_name|
+    lwc = { line: 0, word: 0, byte: 0, name: '' }
+    if FileTest.file?(file_name)
+      file_data = File.read(file_name)
+      lwc[:line] = count_lines(file_data)
+      lwc[:word] = count_words(file_data)
+      lwc[:byte] = File.stat(file_name).size
+    end
+    lwc[:name] = file_name
+    lwc_data << format_lwc(**lwc)
+  end
+  lwc_data
+end
+
+def count_lines(strings)
+  strings.count("\n", "/[^\n]\z/")

```

Comment:
```suggestion
  strings.lines.size
```

とかでもいいかな、と思ったんですが、これだと数字がおかしくなったりします？

Reply:
標準搭載のものと数字が一致しました。
修正しました。